dotcom-rendering/src/web/layouts/FrontLayout.tsx

```diff
@@ -174,6 +185,60 @@ export const FrontLayout = ({ front, NAV }: Props) => {
 						);
 					}
 
+					if (
+						collection.collectionType === 'news/most-popular' &&
+						!isPaidContent &&
+						front.config.switches.mostViewedFronts
+					) {
+						return (
+							<Section
+								key={collection.id}
+								title="Most viewed"
+								showTopBorder={index > 0}
+								padContent={false}
+								verticalMargins={false}
+								url={collection.href}
+								ophanComponentLink={ophanComponentLink}
+								ophanComponentName={ophanName}
+								containerName={collection.collectionType}
+								containerPalette={collection.containerPalette}
+								sectionId={collection.id}
+								showDateHeader={
+									collection.config.showDateHeader
+								}
+								editionId={front.editionId}
+								treats={collection.treats}
+								data-print-layout="hide"
+								element="aside"
+							>
+								<div css={stackBelow('desktop')}>
+									<MostViewedFooter
+										tabs={[
+											{
+												trails: trails.slice(10),
+											},
+										]}
+										sectionName="Most viewed"
+										// TODO: Include mostCommented & mostShared once we have this data in the FE response
+									/>
+									{front.config.switches
+										.extendedMostPopularFronts && (
+										<div
+											css={css`
+												margin: 6px 0 0 10px;
+											`}
+										>
+											<AdSlot
+												position="mostpop"
+												display={format.display}
+											/>
+										</div>
+									)}
+								</div>
+							</Section>
+						);
+					}
```

Comment:
I'm not a big fan of the pattern we have building here of lots of 'large' (no. of lines of code wise) of 'exceptions' building up here, in my opinion it makes the code hard to read, as you can easily have your entire IDE filled with JSX inside the map, and get lost as to where you are. 

In my opinion maps should be relatively concise so you can tell exactly what's going on in them at a glance.

Would this component fit into <DecideContainer>? If not perhaps we could look into some other kind of abstraction in a future PR to make this map more readable?

Reply:
Thanks for raising this. I had a think about this and then saw that there's a way to simplify everything. In [this commit](https://github.com/guardian/dotcom-rendering/pull/5903/commits/85182142bd7b59755381fa139f4b4f6664b71b5b) I've moved this layout logic into `MostViewedLayout` and made that component accept children so that it will work for both articles (where we fetch the data using an api call) and now also fronts (where we render the data directly).

This also has the effect of making the stories for this component simpler

Reply:
> In my opinion maps should be relatively concise so you can tell exactly what's going on in them at a glance

💯 agree on this. Thanks for calling it out!